### PR TITLE
Create conviction rejection form for registrations

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-792

Until we have moved new registrations into the engine and they become transient_registrations, we need to have a slightly separate process for rejecting convictions on registrations.

This PR sets up a version of the conviction rejection form that works with pending new registrations. It doesn't cover making the rejection actually do anything.

We can reuse the ConvictionRejectionForm, but a separate controller, route and view is required.

This is a bit cludgy and duplicates a lot of code, but the intent is that it's very easy to remove once we no longer need it!

---

Evil twin of https://github.com/DEFRA/waste-carriers-back-office/pull/492

### CODE REVIEW

File: app/views/registration_conviction_rejection_forms/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+      <h1 class="heading-large">
+        <%= t(".heading", reg_identifier: @registration.reg_identifier) %>
+      </h1>
+
+      <% if @conviction_rejection_form.errors[:revoked_reason].any? %>
+      <div class="form-group form-group-error">
+      <% else %>
+      <div class="form-group">
+      <% end %>
```
Comment (cintamani) [inline app/views/registration_conviction_rejection_forms/new.html.erb]:
```
      <div class="form-group <%= "form-group-error" if @conviction_rejection_form.errors[:revoked_reason].any? %>">
```

🙇‍♀

File: app/views/registration_conviction_rejection_forms/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        <fieldset id="company_name">
+          <legend class="visuallyhidden">
+            <%= t(".heading", reg_identifier: @registration.reg_identifier) %>
+          </legend>
+
+          <% if @conviction_rejection_form.errors[:revoked_reason].any? %>
+          <span class="error-message"><%= @conviction_rejection_form.errors[:revoked_reason].join(", ") %></span>
```
Comment (cintamani) [inline app/views/registration_conviction_rejection_forms/new.html.erb]: NIT: 
```suggestion
            <span class="error-message"><%= @conviction_rejection_form.errors[:revoked_reason].join(", ") %></span>
```

File: app/presenters/registration_conviction_presenter.rb [hunk 3]
```diff
@@ -8,8 +8,12 @@ def display_actions?
   def begin_checks_path
     Rails.application.routes.url_helpers.registration_convictions_begin_checks_path(reg_identifier)
   end
-
+
```
Comment (irisfaraway) [inline app/presenters/registration_conviction_presenter.rb]:
```suggestion
 
```

File: app/presenters/renewing_registration_conviction_presenter.rb [hunk 4]
```diff
@@ -8,8 +8,12 @@ def display_actions?
   def begin_checks_path
     Rails.application.routes.url_helpers.transient_registration_convictions_begin_checks_path(reg_identifier)
   end
-
+
```
Comment (irisfaraway) [inline app/presenters/renewing_registration_conviction_presenter.rb]:
```suggestion
 
```

File: app/presenters/registration_conviction_presenter.rb [hunk 5]
```diff
@@ -8,8 +8,12 @@ def display_actions?
...
```
Comment (irisfaraway) [inline app/presenters/registration_conviction_presenter.rb]:
```suggestion

```

File: app/presenters/renewing_registration_conviction_presenter.rb [hunk 6]
```diff
@@ -8,8 +8,12 @@ def display_actions?
...
```
Comment (irisfaraway) [inline app/presenters/renewing_registration_conviction_presenter.rb]:
```suggestion

```

**Pull request merged**